src/Microsoft.AspNet.Http.Extensions/HeaderDictionaryExtensions.cs

```diff
@@ -23,7 +24,7 @@ public static void Append(this IHeaderDictionary headers, string key, StringValu
         /// </summary>
         /// <param name="key">The header name.</param>
         /// <param name="values">The header values.</param>
-        public static void AppendCommaSeparatedValues(this IHeaderDictionary headers, string key, params string[] values)
+        public static void AppendCommaSeparatedValues(this IDictionary<string, StringValues> headers, string key, params string[] values)
```

Comment:
@Tratcher raised a good point about the contract of ```IDictionary``` being throw on ```Get``` so a different interface would likely be better that returns default(TValue) on ```Get```

Wouldn't be an extra burden on Kestrel's HeaderDict as you don't actually wrap a dictionary and are implementing the interface in this manner. Would also allow a IReadOnly->IReadWrite interface hierarchy should it be desired which is not possible with IDictionary and IReadOnlyDictionary (see https://github.com/aspnet/HttpAbstractions/pull/411#issuecomment-142065105)